src/client/components/trackDuration/trackDuration.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react';
+import 'react-rangeslider/lib/index.css';
+import './trackDuration.css';
+import PropTypes from 'prop-types';
+import Slider from 'react-rangeslider';
+import { connect } from 'react-redux';
+import VolumeComponent from '../volumeComponent/volumeComponent';
+import SwitcherSound from '../switcherSound/switcherSound';
+
+class TrackDuration extends Component {
+    state = {
+      duration: null,
+      currentTime: null,
+      playing: false,
+      startPlayTime: null,
+      onToggle: true,
+    }
+
+    static getDerivedStateFromProps(props, state) {
+      const { audioData: { loading, sound, trackName } } = props;
+      if (sound && sound.sourceNode) {
+        if (!state.startPlayTime && sound.playing) {
+          return {
+            startPlayTime: new Date(),
+            duration: Math.round(sound.sourceNode.buffer.duration),
+            playing: sound.playing,
+            trackName,
+          };
+        }
+        if (loading) {
+          return {
+            playing: false,
+          };
+        }
+        return {
+          duration: Math.round(sound.sourceNode.buffer.duration),
+          playing: sound.playing,
+          trackName,
+        };
+      }
+      return null;
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+      if (!prevState.playing && this.state.playing) {
+        this.setState({ startPlayTime: new Date(new Date() - prevState.currentTime * 1000) });
+      }
+      if (this.state.startPlayTime) {
+        this.calculateCurrentTime();
+      }
+      if (prevState.trackName !== this.state.trackName) {
+        this.setState({ startPlayTime: new Date() });
+      }
+    }
+
+    formatingSongTime = (numberOfSecs) => {
+      const seconds = numberOfSecs % 3600;
+      let min = (Math.floor(seconds / 60)).toFixed();
+      let sec = (seconds % 60).toFixed();
+
+      min = String(min).padStart(2, '0');
+      sec = String(sec).padStart(2, '0');
+      return `${min}:${sec}`;
+    }
+
+    calculateCurrentTime = () => {
+      const currentTime = new Date();
+      const currentDifference = Math.round(
+        (currentTime.getTime() - this.state.startPlayTime.getTime()) / 1000,
+      );
+      const { audioData: { sound } } = this.props;
+      if (this.state.currentTime >= this.state.duration) {
+        sound.stop();
+        sound.playing = false;
+        this.setState({
+          currentTime: 0,
+          playing: false,
+        });
+      } else if (currentDifference !== this.state.currentTime && this.state.playing) {
+        this.setState({ currentTime: currentDifference },
+          () => this.calculateCurrentTime());
+      } else {
+        setTimeout(this.calculateCurrentTime, 1000);
+      }
+    }
+
+    setPauseOnSound = () => {
```

Comment:
you can rewrite this function and `setPlayOnSound` to use only one function, for example `togglePlay`
on click it will play sound on second click it will pause